spring-kafka/src/main/java/org/springframework/kafka/listener/BatchInterceptor.java

```diff
@@ -61,4 +61,29 @@ default void success(ConsumerRecords<K, V> records, Consumer<K, V> consumer) {
 	default void failure(ConsumerRecords<K, V> records, Exception exception, Consumer<K, V> consumer) {
 	}
 
+	/**
+	 * Called before consumer is polled.
+	 * <p>
+	 * It can be used to set up thread-bound resources which will be available for the
+	 * entire duration of the record processing e.g. logging with MDC.
+	 * </p>
+	 * @param consumer the consumer.
+	 * @since 2.8
+	 */
+	default void beforePoll(Consumer<K, V> consumer) {
+	}
+
+	/**
+	 * Called after records were processed by listener and error handler.
+	 * <p>
+	 * It can be used to clear thread-bound resources which were set up in {@link #beforePoll(Consumer)}.
+	 * This is the last method called by the {@link MessageListenerContainer} before the next record
+	 * processing cycle starts.
+	 * </p>
+	 * @param consumer the consumer.
+	 * @since 2.8
+	 */
+	default void afterRecordsProcessed(Consumer<K, V> consumer) {
```

Comment:
Don't we need to have a `ConsumerRecords<K, V> records` context over here as well?
You maybe would like to know for what batch is this stateless interceptor.
See an `ExecutorChannelInterceptor` contract in Spring Messaging for example.

Reply:
That's a good suggestion.